src/oaklib/utilities/lexical/lexical_indexer.py

```diff
@@ -406,14 +407,18 @@ def apply_transformation(term: str, transformation: LexicalTransformation) -> st
         )
 
 
-def apply_synonymizer(term: str, rules: List[Synonymizer]) -> str:
+def apply_synonymizer(term: str, rules: List[Synonymizer]) -> Tuple:
```

Comment:
always include as precise types as possible

better:

`Tuple[bool, str, str]`

even better, something like:

`Tuple[bool, str, SYNONYM_QUALIFIER_TYPE]`

have a look at existing tuple declarations in OAK

Also add a docstring here. In general docstrings should go on public methods, and *especially* central ones
like this that are really necessary for understanding synonymizer

Reply:
Done